File: 6_kyu/Who_likes_it.py
def likes(names):
    l=len(names)
    if l is 0:
        return "no one likes this"
    elif l is 1:      
        return f"{names[0]} likes this"
    elif l is 2:
        return f"{names[0]} and {names[1]} like this"
    elif l is 3:
        return f"{names[0]}, {names[1]} and {names[2]} like this"
    else:
        return f"{names[0]}, {names[1]} and {len(names)-2} others like this"

File: 6_kyu/Who_likes_it.jl
module Kata
  export likes
  
  "Returns 'like' statement including the firsts of the given names."
  function likes(names::Vector{String}) :: String
    len = length(names)
    len == 0 && return "no one likes this"
    len == 1 && return "$(names[1]) likes this"
    len == 2 && return "$(names[1]) and $(names[2]) like this"
    len == 3 && return "$(names[1]), $(names[2]) and $(names[3]) like this"
    "$(names[1]), $(names[2]) and $(len-2) others like this"
  end
end
